### PR TITLE
 Allow specifiying variable pointers in requirements.

### PR DESCRIPTION
**Pull request by dj2**
This CL adds the two requirements:
 * VariablePointerFeatures.variablePointers
 * VariablePointerFeatures.variablePointersStorageBuffer

 Variable pointers requires VK_KHR_get_physical_properties2 and the
 variable pointer extension are both requested.

### CODE REVIEW

**Comment by dj2** [comment]
@alan-baker

File: samples/config_helper_vulkan.cc [hunk 1]
```diff
@@ -757,14 +791,33 @@ amber::Result ConfigHelperVulkan::CreateDeviceWithFeatures1(
...
+    // No dot means this is a features1 feature.
+    if (feature.find_first_of('.') == std::string::npos) {
+      feature1_names.push_back(feature);
+      continue;
+    }
+
+    if (feature == kVariablePointers)
+      variable_pointers_feature_.variablePointers = VK_TRUE;
+    else if (feature == kVariablePointersStorageBuffer)
+      variable_pointers_feature_.variablePointersStorageBuffer = VK_TRUE;
```
Comment (dj2) [inline samples/config_helper_vulkan.cc]: Comment from @alan-baker:

"The spec says if you turn on VK_KHR_variable_pointers you MUST support variablePointersStorageBuffer so I'd default that to true if the extension or VariablePointerFeatures.variablePointers is mentioned"

Do we want the implicit setting to true, or should we make the script have to be explicit?

File: samples/config_helper_vulkan.cc [hunk 2]
```diff
@@ -183,7 +187,7 @@ amber::Result NamesToVulkanFeatures(
     } else if (name == "inheritedQueries") {
       features->inheritedQueries = VK_TRUE;
     } else {
-      return amber::Result("Sample: Unknown Vulkan feature");
+      return amber::Result("Sample: Unknown Vulkan feature: " + name);
```
Comment (dneto0) [inline samples/config_helper_vulkan.cc]: this compiles!?
TIL

I expected the LHS of the + to be a std::string

File: samples/config_helper_vulkan.cc (hunk 1 above)
Comment (dneto0) [reply samples/config_helper_vulkan.cc]: I'm ok with forcing the script to be explicit.  I think.

Comment (alan-baker) [reply samples/config_helper_vulkan.cc]: What about VariablePointerFeatures.variablePointers implying both variables?

File: samples/config_helper_vulkan.cc (hunk 2 above)
Comment (dj2) [reply samples/config_helper_vulkan.cc]: amber::Result takes a `const std::string&` as input. So, it's probably converting the first const char* to std::string then doing the add.

File: samples/config_helper_vulkan.cc (hunk 1 above)
Comment (dj2) [reply samples/config_helper_vulkan.cc]: I'm leaning towards explicit so yo have to specify both. It makes the test a lot clearer and removes spooky action at a distance.

**Pull request merged**